tests/performance_tests/performance_test_basic.py

```diff
@@ -209,7 +210,11 @@ def runTpsTest(self) -> bool:
             ])
 
         # Get stats after transaction generation stops
-        self.data.ceaseBlock = self.waitForEmptyBlocks(self.validationNode, self.emptyBlockGoal) - self.emptyBlockGoal + 1
+        trxSent = {}
+        log_reader.scrapeTrxGenTrxSentDataLogs(trxSent, self.trxGenLogDirPath, self.quiet)
+        blocksToWait = ceil(self.expectedTransactionsSent / min(4000, 0.45 * self.targetTps))
```

Comment:
Why not use a simple calculation based on `self.testTrxGenDurationSec` instead?  Could maybe wait `2 * self.testTrxGenDurationSec` worth of blocks?  Blocks are produced every half second, so it would be more like: `2 * 2 * self.testTrxGenDurationSec` to allow it to wait double the test generation duration.

Reply:
Might make it easier to understand as it would just be a multiplier against a duration in seconds instead of transaction and tps based calculations.